eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs

```diff
@@ -201,10 +207,16 @@ spoAcceptedRatio
       accumStake (!yes, !abstain) poolId distr =
         let CompactCoin stake = individualTotalPoolStake distr
             vote = Map.lookup poolId gasStakePoolVotes
+            res =
+              Map.lookup poolId rePoolParams >>= \d -> Map.lookup ((raCredential . ppRewardAccount) d) reDelegatees
          in case vote of
-              Nothing
-                | HardForkInitiation {} <- pProcGovAction -> (yes, abstain)
-                | otherwise -> (yes, abstain + stake)
+              Nothing -> case res of
+                Just DRepAlwaysNoConfidence
+                  | NoConfidence {} <- pProcGovAction -> (yes + stake, abstain)
+                  | otherwise -> (yes, abstain)
+                Just DRepAlwaysAbstain -> (yes, abstain + stake)
+                -- TODO: figure out if other cases need special care
+                _ -> (yes, abstain)
```

Comment:
We need to make sure that:
* we do not change the behavior of `HardForkInitiation`
* we preserve the bootstrap behavior
* we have access to protocol version, which we probably can also stick into `RatifyEnv`
```suggestion
              Nothing
                | HardForkInitiation {} <- pProcGovAction -> (yes, abstain)
                | bootstrapPhase pv -> (yes, abstain + stake)
                | otherwise -> case res of
                  Just DRepAlwaysNoConfidence
                    | NoConfidence {} <- pProcGovAction -> (yes + stake, abstain)
                  Just DRepAlwaysAbstain -> (yes, abstain + stake)
                  _ -> (yes, abstain)
```